run-build-functions.sh

```diff
@@ -323,6 +332,24 @@ install_dependencies() {
       echo "Error installing pip dependencies"
       exit 1
     fi
+  elif [ -f Pipfile ]
+  then
+    echo "Installing dependencies from Pipfile"
+    if $HOME/python$PIPENV_RUNTIME/bin/pipenv install
+    then
+      echo "Pipenv dependencies installed"
+      if source $($HOME/python$PIPENV_RUNTIME/bin/pipenv --venv)/bin/activate
```

Comment:
fwiw, doing this instead of `pipenv shell` and `pipenv run` means `.env` loading is not supported, which could break some `Pipfile` if it is expecting variables from `.env` to be present.

https://github.com/pypa/pipenv/blob/master/docs/advanced.rst#-automatic-loading-of-env
https://github.com/pypa/pipenv/blob/master/docs/advanced.rst#-support-for-environment-variables

Note that it isnt even possible to use `.env` vars during `pipenv install`.  see https://github.com/pypa/pipenv/issues/1906#issuecomment-378635177

As a result, probably a good idea to set `PIPENV_DONT_LOAD_ENV` and document that .env isnt used by Netlify.

Reply:
Thanks for the feedback! It seems to me that ENV variables are meant to be handled some other way on production. E.g. using https://www.netlify.com/docs/continuous-deployment/#build-environment-variables.

Based on discussion in https://github.com/pypa/pipenv/issues/773#issuecomment-333246959 `pipenv shell` is meant only for development and using `source .../bin/activate` is OK for production.